atmcfg/atmcfg.go

```diff
@@ -464,3 +504,19 @@ func reclaimByShardName(out *opsmngr.AutomationConfig, name, lastCompact string)
 		// compact doesn't run on mongoses
 	}
 }
+
+func splitHostnameAndPort(hostnameAndPort string) (hostname string, port int, err error) {
+	hostnameAndPortSlice := strings.Split(hostnameAndPort, ":")
+
+	if len(hostnameAndPortSlice) == 1 {
```

Comment:
[nit] what happens if length is 3?
```suggestion
	if len(hostnameAndPortSlice) != 2 {
```

Reply:
but see my other comment where we may not need this